### ISSUE TITLE
Google Services say - Browser is not supported

### ISSUE THREAD

**Issue by leoheck** [getferdi/ferdi#1390]
Since some days ago, I started to see this message.
![image](https://user-images.githubusercontent.com/1277920/117866550-a71ff180-b26d-11eb-926f-e1c9de0b26fc.png)

It mostly happens after login. Google Chat/Calendar keeps login out for no reason.

I am using Franz 5.6.0-nightly.15 on Ubuntu 21.04
REloading the service a couple of times seems to work.

**Comment by vraravam** [comment]
The latest code for the recipes are now all merged. Please refer to the brand new [self-help documentation](https://drive.google.com/drive/folders/1mJmKbhpVHdb066IHlrdYiD8TG9wlmuBv) - which might help to solve the issue.

**Comment by vraravam** [comment]
@leoheck - please check if upgrading the recipes in your installation helps fix the above issue. If it does, please close this issue.

**Comment by leoheck** [comment]
>  self-help documentation 

Well it is not pretty clear to me what I have to do. 
How can I upgrade recipes? What the recipes are?

**Comment by vraravam** [comment]
@leoheck - sorry, I had pasted the wrong link - now the link for the `self-help documentation` is corrected in the previous comment. Please check if you can follow that.

**Comment by vraravam** [comment]
@leoheck : If the workaround to update to the latest recipes worked, please close this bug ticket.

**Comment by leoheck** [comment]
Sure! @vraravam I cannot open the self-help documentation. Can you check if the URL still works?

**Comment by vraravam** [comment]
Sorry - this PR has now been merged into the main Ferdi repo: https://github.com/getferdi/ferdi/#troubleshooting-recipes-self-help

**Comment by leoheck** [comment]
So, lazy way

```bash
#!/bin/bash

cd ~/.config/Ferdi/recipes

git clone https://github.com/getferdi/recipes.git

dirs=$(find . -mindepth 1 -maxdepth 1 -type d \( ! -name recipes \) \( ! -name temp \) \( ! -name ".*" \))

mkdir -p .old_recipes

IFS=$'\n'

for i in ${dirs}; do
	mv -f $i .old_recipes
	ln -sf recipes/recipes/$i $i
done
```

So after that, it is just a matter of updating the repo
``` 
cd ~/.config/Ferdi/recipes/recipes
git pull
```

But it didn't work. Actually, I don't have services anymore.
=/.

**Comment by vraravam** [comment]
Can you please blast away your `~/.config/Ferdi/` directory? I suspect that your script up above has placed `recipes` in a different nested location. (In the git repo, its `recipes/recipes`, but, in the runtime profile location there's only a single level of nesting in `recipes`. This is just a guess since I can't run your script against my install on a mac,

**Comment by leoheck** [comment]
Hey @vraravam, sure I can. Lets see if it works.

**Comment by leoheck** [comment]
Cool, this was good. I was able to log in and load back my stuff, thanks.

Now getting back to the new recipes. Should I update them again or I am going to have new recipes?

Well, I am testing this now.. just to check.
Running the script 
![image](https://user-images.githubusercontent.com/1277920/118203070-a9cb4400-b431-11eb-9832-aac629134d40.png)

Then, the same problem.
This might be a problem with the symlinks I am using on this script.

I will test moving directories instead... 
Let me blast the whole folder again to start it all over again.
Now I have folders instead of symlinks
![image](https://user-images.githubusercontent.com/1277920/118203318-2e1dc700-b432-11eb-9368-645e158ce477.png)

Here I am showing that the bkp of the old and working recipes are actually different than the new ones. This is the diff of the whole whastapp folder, for instance.
![image](https://user-images.githubusercontent.com/1277920/118203403-5e656580-b432-11eb-9016-e423b0972ebd.png)

Same problem happens. Zero services working.
![image](https://user-images.githubusercontent.com/1277920/118203515-9d93b680-b432-11eb-8a99-cd2cf0df6a18.png)

Are you sure it is just a matter of replacing these files with the ones from here  https://github.com/getferdi/recipes/tree/master/recipes ?

**Comment by leoheck** [comment]
Just to make sure the script was not messing up with anything as you pointed out. 
I moved things by hand, old services were renamed with `_old` prefix. 
New services were copied from `recipes/recipes/`
![image](https://user-images.githubusercontent.com/1277920/118204034-b18be800-b433-11eb-9cb5-8f5d97237b66.png)

**Comment by vraravam** [comment]
What's the version of Ferdi that you are running @leoheck ?

**Comment by vraravam** [comment]
Also, (just a guess), but can you move your old recipes completely out? (Maybe the old and new are conflicting with some kind of recipe_id field.

**Comment by leoheck** [comment]
The version is 5.6.0-nightly-15

Sure, let me purge these old folders.
I removed everything that was duplicated, including the cloned recipes.

Alright, it worked.

Here is the updated script if you need it for someone else.
https://gist.github.com/leoheck/40ae0e98ddca5bb102c2dffbaba3cb5b

Thanks for the idea dude!

**Comment by vraravam** [comment]
@leoheck - since the problem is resolved, I'm closing this ticket.

**Comment by Soetens** [comment]
I'm still having this problem, but with Windows and not sure how to fix this.

I can;t find where the recipes are stored.
C:\Users\xxxxx\AppData\Local\Programs\ferdi\recipes\
doesnt exists.

But to be sure I coped all the recipes from the recipe git to
C:\Users\xxxxx\AppData\Local\Programs\ferdi\recipes\

But nothing changed.

**Comment by vraravam** [comment]
hi @Soetens , I would suggest to go to your `Preferences` section, click on the `Your Services` on the top left, click on one of the configured services in the right, scroll down. You will find 3 buttons stating open XXX - you can click on any one of them - and they will open windows explorer in the correct location. Once this is done, close Ferdi, and within windows explorer - perform the steps that I had outlined in the `README`. Once complete, restart Ferdi.

**Comment by Soetens** [comment]
That indeed helped a lot! For future Windows users with the problem replace the recipe in
`C:\Users\xxxxx\AppData\Roaming\Ferdi\recipes\gmail`

**Comment by vraravam** [comment]
Thanks for the heads up @Soetens  - and glad that it worked out for you. Can we now close this bug?

PS: I just realized that this is already documented in the main `README` section that I had linked above.

**Comment by Soetens** [comment]
It works for the existing Gmail service. But when I try to add a new Gmail service I see a reload button. After pressing that it just reloads the current active service (not the gmail service, but a different one that is open).
I have to quit Ferdi and start it again.

At this point adding a new Gmail service does not seem to be possible.

**Comment by vraravam** [comment]
The 'Reload' functionality is quite different. afaik, its useful to refresh the service webview and start as-if restarting Ferdi (with some exceptions to that assumption).

For adding a new Gmail service, when you click the "Add Service" button, you can choose "gmail", and then give it a different name. That should allow you to add more than 1 instance of the same service.

btw - the recipe/service configs that you updated from the git repo - that's just a template for each service. So, there will only be 1 instance of the "gmail" folder in your file explorer. Hope that clarifies the difference.

**Comment by Soetens** [comment]
But when I press I directly get the reload button, I don't get the modal to add services.
![image](https://user-images.githubusercontent.com/32106310/118290185-6d055880-b4d6-11eb-97af-56791933b60a.png)
